autogpt_platform/backend/backend/integrations/oauth/google.py

```diff
@@ -35,36 +47,74 @@ def get_login_url(self, scopes: list[str], state: str) -> str:
         )
         return authorization_url
 
-    def exchange_code_for_tokens(self, code: str) -> OAuth2Credentials:
-        flow = self._setup_oauth_flow(None)
+    def exchange_code_for_tokens(
+        self, code: str, scopes: list[str]
+    ) -> OAuth2Credentials:
+        logger.debug(f"Exchanging code for tokens with scopes: {scopes}")
+
+        # Use the scopes from the initial request
+        flow = self._setup_oauth_flow(scopes)
         flow.redirect_uri = self.redirect_uri
-        flow.fetch_token(code=code)
 
-        google_creds = flow.credentials
-        username = self._request_email(google_creds)
+        logger.debug("Fetching token from Google")
 
-        # Google's OAuth library is poorly typed so we need some of these:
-        assert google_creds.token
-        assert google_creds.refresh_token
-        assert google_creds.expiry
-        assert google_creds.scopes
```

Comment:
So we can't assume these are always present on `google_creds`? Why not?

Reply:
Assert doesn’t always do what you’d think. If we need to check this, we need to specifically check and raise.

Reply:
adding this back

Reply:
> I read this as just telling the type checked they WILL exist.

Yes, that's what they were for.